File: Day01_Music/app/src/main/java/day01/nguyenhongphat0/day01_music/PlayMusicService.java
package day01.nguyenhongphat0.day01_music;

import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.IBinder;

public class PlayMusicService extends Service {
    MediaPlayer mp;

    public PlayMusicService() {
    }

    @Override
    public IBinder onBind(Intent intent) {
        // TODO: Return the communication channel to the service.
        return null;
    }

    @Override
    public void onCreate() {
        mp = MediaPlayer.create(getApplicationContext(), R.raw.songsao);
        super.onCreate();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        mp.start();
        return Service.START_STICKY_COMPATIBILITY;
    }

    @Override
    public void onDestroy() {
        mp.release();
        super.onDestroy();
    }
}

File: kotlin/Day01/app/src/main/java/day01/nguyenhongphat0/day01/PlayMusicService.kt
package day01.nguyenhongphat0.day01

import android.app.Service
import android.content.Intent
import android.media.MediaPlayer
import android.os.IBinder

class PlayMusicService : Service() {
    lateinit var mp: MediaPlayer

    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onCreate() {
        super.onCreate()
        mp = MediaPlayer.create(this, R.raw.songsao)
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        mp.start()
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        super.onDestroy()
        mp.release()
    }
}
